src/test/java/com/yannbriancon/interceptor/QueryCountTest.java

```diff
@@ -45,4 +52,24 @@ void queryCount_isOkWhenSaveQueryIsExecutedBeforeStartingTheCount() {
 
         assertThat(hibernateQueryInterceptor.getQueryCount()).isEqualTo(1);
     }
+
+    @ParameterizedTest
+    @MethodSource("prepareQueryCountWithStartsWithFilterData")
+    void queryCountWithStartsWithFilter(String filter, int expectedQueryCount) {
```

Comment:
Good test!
Please add tests to catch the bugs explained in my comments.

Reply:
Thx. Do you mean i should do a multithreaded test with TestNG for example?